src/thunderbots/software/network_input/backend.cpp

```diff
@@ -1,149 +1,218 @@
+                                   ssl_field_lines["LeftPenaltyStretch"].p1().y());
+    Point defense_width_p2 = Point(ssl_field_lines["LeftPenaltyStretch"].p2().x(),
+                                   ssl_field_lines["LeftPenaltyStretch"].p2().y());
+    double defense_width =
+        (defense_width_p1 - defense_width_p2).len() * METERS_PER_MILLIMETER;
+
+    Field field = Field(field_length, field_width, defense_length, defense_width,
+                        goal_width, boundary_width, center_circle_radius);
+    return field;
+}
+
+Ball Backend::getFilteredBallData(const std::vector<SSL_DetectionFrame> &detections)
+{
+    std::vector<SSLBallDetection> ball_detections = std::vector<SSLBallDetection>();
+
+    for (const auto &detection : detections)
+    {
+        for (const SSL_DetectionBall &ball : detection.balls())
         {
-            std::vector<SSLBallData> ball_detections = std::vector<SSLBallData>();
-            for (const SSL_DetectionBall &ball : detection.balls())
-            {
-                // Convert all data to meters and radians
-                SSLBallData ball_data;
-                ball_data.position   = Point(ball.x() * METERS_PER_MILLIMETER,
-                                           ball.y() * METERS_PER_MILLIMETER);
-                ball_data.confidence = ball.confidence();
-                ball_data.timestamp  = detection.t_capture();
-                ball_detections.push_back(ball_data);
-            }
-
-            FilteredBallData filtered_ball_data =
-                ball_filter.getFilteredData(ball_detections);
-            thunderbots_msgs::Ball ball_msg =
-                MessageUtil::createBallMsgFromFilteredBallData(filtered_ball_data);
-            return ball_msg;
+            // Convert all data to meters and radians
+            SSLBallDetection ball_detection;
+            ball_detection.position =
+                Point(ball.x() * METERS_PER_MILLIMETER, ball.y() * METERS_PER_MILLIMETER);
+            ball_detection.confidence = ball.confidence();
+            ball_detection.timestamp  = Timestamp::fromSeconds(detection.t_capture());
+            ball_detections.push_back(ball_detection);
         }
     }
 
-    return std::nullopt;
+    Ball updated_ball_state = ball_filter.getFilteredData(ball_state, ball_detections);
+    ball_state              = updated_ball_state;
+
+    return ball_state;
 }
 
-std::optional<thunderbots_msgs::Team> Backend::getFilteredFriendlyTeamMsg(
-    const SSL_WrapperPacket &packet)
+Team Backend::getFilteredFriendlyTeamData(std::vector<SSL_DetectionFrame> detections)
 {
-    if (packet.has_detection())
-    {
-        const SSL_DetectionFrame &detection                = packet.detection();
-        std::vector<SSLRobotData> friendly_team_robot_data = std::vector<SSLRobotData>();
+    std::vector<SSLRobotDetection> friendly_robot_detections =
+        std::vector<SSLRobotDetection>();
```

Comment:
nit; but I'd avoid duplicate type declarations by making this 
```
auto friendly_robot_detections = std::vector<SSLRobotDetection>();
```

Reply:
fixed